Assets/_Scripts/BattleController.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+public enum BattlePhase {
+    PLAYER,
+    ENEMY
+
+};
+
+
+public class BattleController : MonoBehaviour
+{
+    public List<RectTransform> battleOptionsUI;
+    public RectTransform selectionCursor;
+    public Vector2 selectionCursorOffset = new Vector2(0, 0);
+
+    public float maxTimeBeforeAction = 15;
+    public float playerActionCounter {get; set;}
+
+    public RectTransform ActionMenu;
+    public RectTransform CommentaryMenu;
+    public Text CommentaryText;
+
+
+    List<Player> _players;
+    Player _heroPlayer;
+
+
+    bool startActions = false;
+
+
+    void Start() {
+        _players = GameManager.Instance.party.GetPlayersInPosition();
+        _heroPlayer = _players[0];
+
+        Debug.Log("Players: " + _players.Count);
+        for (int i = 0; i < _players.Count; i++) {
+            Debug.Log(_players[i].Name);
+        }
+
+        // TODO: Waves
+        GameManager.Instance.GenerateEnemyList();
```

Comment:
We could store enemies like players every turn. That way we only need to interact with game manager at the beginning of the turn.